### DISCUSSION TITLE
Repeating Code

### DISCUSSION THREAD

**Discussion by Guilherme-Cassel** [QuestPDF/QuestPDF#768] [Q&A] [answered]
It looks like some code parts of QuestPDF are supposed to be repeated every item, but look at this example I made here

![image](https://github.com/QuestPDF/QuestPDF/assets/150675837/e19c8c87-72ec-4128-926a-1a208132a103)

am I doing it right? 

you can see I have to repeat ".Border(1).AlignCenter().AlignMiddle().FontSize(11)" for every item on the row for it to work.
it doesnt feel right but I'm here to see any other opinion about this, and if there is a way to overcome this that I'm not aware of, please share 🥰

**Comment by girlpunk** [comment, marked as answer]
As a suggestion:

```csharp
public static IContainer MyTableCell(this IContainer container) => container
    .Border(1)
    .AlignCenter()
    .AlignMiddle();

public static void MyTextCell(this IContainer container, string text) => container
    .MyTableCell()
    .Text(text)
    .FontSize(11);

private void TableHeader(IContainer container)
{
    container
        .Border(1)
        .Background("a4efc4")
        .Row(row =>
        {
            row.RelativeItem(3).MyTextCell("Produto");
            row.RelativeItem(17).MyTextCell("Descrição");
            row.RelativeItem(4).MyTextCell("Qntd");
        });
}
```

**Reply by Guilherme-Cassel** [reply]
My Report Class is not Static so I cant make extension methods. I could do put in another class tough (which is the correct way) but each report has a different cell formatation so I would need to make repeated extension methods and we get back to the primary problem :/

By now I guess I'll just accept my fate of repeating this part of the code, I was thinking in someting like "container.Format = aligncenter().alignmiddle().fontsize(11)" you know, then all items on this container would have the same formatation, but its not a thing yet

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
